pandas/core/frame.py

```diff
@@ -4538,7 +4539,7 @@ def _clear_item_cache(self) -> None:
 
     def _get_item_cache(self, item: Hashable) -> Series:
         """Return the cached item, item represents a label indexer."""
-        if using_copy_on_write():
+        if using_copy_on_write() or warn_copy_on_write():
             loc = self.columns.get_loc(item)
             return self._ixs(loc, axis=1)
```

Comment:
I needed to make a decision here about what to do with the item cache in the warning mode.

In the end, the current version of the PR updates the warning mode to also _not_ populate and use the item cache, like for Cow mode. 

This will cause some behaviour changes when enabling the warning mode. But it also avoids a bunch of otherwise false-positive warnings. So it's a trade-off.

Examples that would otherwise give false positives:

```python
df =  ..
df["col"].mean()
# now mutating `df` would trigger CoW / raise warning because the df["col"] series is cached
```

or whenever we access columns under the hood in a method, like `df2 = df.astype({..})`, even when df2 fully consists of copied data because there was an actual cast, `df` will have referenced because of the item cache.

Reply:
Fine by me